lib/bdd/rspec/example_group_steps.rb

```diff
@@ -0,0 +1,78 @@
+module Bdd
+  module RSpec
+    module ExampleGroupSteps
+
+      module ClassMethods
+        def define_bdd_step(*names)
+          bdd_steps.concat(names)
+          names.each do |name|
+            define_method(name) do |msg, &block|
+              bdd_step([name, msg], &block)
+            end
+          end
+        end
+        def bdd_steps
+          @bdd_steps
+        end
+      end
+
+      def self.included(base)
+        # add the `define_bdd_step` and `bdd_steps` class methods
+        base.extend ClassMethods
+        # initialize empty bdd_steps
+        base.instance_variable_set(:@bdd_steps, [])
+        # step needs access to the @bdd_steps - but it is not visible when extend is used, especially not as @@bdd_steps
+        base.send(:define_method, :bdd_steps) { base.bdd_steps }
+      end
+
+    private
+
+      def bdd_step(prefix_and_text, &block)
+        unless using_bdd_formatter?
+          yield if block_given?
+          return
+        end
+
+        if
+          ::RSpec.current_example.metadata[:bdd_prefix_max_length].nil?
+        then
+          ::RSpec.current_example.metadata[:bdd_prefix_max_length] = bdd_steps.map(&:length).max
+        end
+
+        if
+          block_given?
+        then
+          if
+            @is_during_rspec_step
+          then
+            yield
+          else
+            bdd_step_messages << prefix_and_text
+            @is_during_rspec_step = true
+            yield
+            @is_during_rspec_step = false
+          end
+
+        elsif
+          prefix_and_text[1] == :bdd
```

Comment:
I have investigated, the purpose of that is to allow 'renaming' a block

this looks simple
```ruby
When("I sign in") {  ... }
```

this looks simple too
```ruby
def when_I_sign_in
When("I sign in") {  ... }
end
```
```ruby
Given "I have signed in" do
  when_I_sign_in
end


# is the same as
Given "I have signed in" do
  When("I sign in") {  ... }
end
```

In the above code, Given should override When.